app/controllers/settings/general.js

```diff
@@ -109,6 +111,8 @@ export default Controller.extend({
             // set a new random password when isPrivate is enabled
             if (isPrivate && this.get('settings.hasDirtyAttributes')) {
                 this.get('settings').set('password', randomPassword());
+            } else {
+                this.get('settings').set('password', this._initialPasswordVal);
```

Comment:
Rather than storing the initial value on the controller and requiring a change on the route (which shouldn't really be involved in this) we can instead use Ember Data's change tracking to reset the value and keep all the logic in one place:

*services/settings.js*
```javascript
    changedAttributes() {
        return this.get('content').changedAttributes();
    },
```

*controllers/settings/general.js*
```javascript
        toggleIsPrivate(isPrivate) {
            let settings = this.get('settings');

            settings.set('isPrivate', isPrivate);
            settings.get('errors').remove('password');

            let changedAttrs = settings.changedAttributes();

            // set a new random password when isPrivate is enabled
            if (isPrivate && changedAttrs.isPrivate) {
                settings.set('password', randomPassword());

            // reset the password when isPrivate is disabled
            } else if (changedAttrs.password) {
                settings.set('password', changedAttrs.password[0]);
            }
        },
```